src/Workspaces/CSharp/Portable/FindSymbols/CSharpDeclaredSymbolInfoFactoryService.cs

```diff
@@ -492,5 +492,75 @@ private static string GetSimpleTypeName(SimpleNameSyntax name)
         private bool IsExtensionMethod(MethodDeclarationSyntax method)
             => method.ParameterList.Parameters.Count > 0 &&
                method.ParameterList.Parameters[0].Modifiers.Any(SyntaxKind.ThisKeyword);
+
+        public override string GetRootNamespace(CompilationOptions compilationOptions)
+            => string.Empty;
+
+        public override bool TryGetTargetTypeName(SyntaxNode node, out string targetTypeName)
+        {
+            if (node is MethodDeclarationSyntax methodDeclaration && IsExtensionMethod(methodDeclaration))
+            {
+                targetTypeName = GetTypeName(methodDeclaration.ParameterList.Parameters[0].Type);
+                return true;
+            }
+
+            targetTypeName = null;
+            return false;
+
+            static string GetTypeName(TypeSyntax typeNode)
+            {
+                switch (typeNode)
+                {
+                    case IdentifierNameSyntax identifierNameNode:
+                        return identifierNameNode.Identifier.Text;
+
+                    case GenericNameSyntax genericNameNode:
+                        var name = genericNameNode.Identifier.Text;
+                        var arity = genericNameNode.Arity;
+                        return arity == 0
+                            ? name
+                            : name + GetMetadataAritySuffix(arity);
+
+                    case PredefinedTypeSyntax predefinedTypeNode:
+                        return GetSpecialTypeName(predefinedTypeNode);
+
+                    case AliasQualifiedNameSyntax aliasQualifiedNameNode:
+                        return GetTypeName(aliasQualifiedNameNode.Name);
+
+                    case QualifiedNameSyntax qualifiedNameNode:
+                        return GetTypeName(qualifiedNameNode.Right);
+
+                    case NullableTypeSyntax nullableNode:
+                        return GetTypeName(nullableNode.ElementType);
```

Comment:
nullable should be explained why it's ok.  i think it's likely to be fine, but it still needs explanation that this has been thought through.

Reply:
added comment in code